app/models/user.rb

```diff
@@ -27,8 +27,8 @@ def after_confirmation
 
   def assign_role_and_admin_id
     if (role.nil? && admin_id.nil?)
-      user = User.find_by(role: "Admin", admin_id: id)
-      update_columns(role: "Admin", admin_id: id) if user.nil?
+      user = User.find_by(role: 3, admin_id: id)
```

Comment:
Hardcoded id. It is high possibility, that it break the thing. Use enum constants only